app/src/main/kotlin/com/simplemobiletools/smsmessenger/activities/ThreadActivity.kt

```diff
@@ -710,6 +713,30 @@ class ThreadActivity : SimpleActivity() {
         setupScheduleSendUi()
     }
 
+    private fun askForExactAlarmPermissionIfNeeded(callback: () -> Unit = {}) {
+        if (isSPlus()) {
+            val alarmManager: AlarmManager = getSystemService(ALARM_SERVICE) as AlarmManager
+            when {
```

Comment:
this should be an if/else, not when/else

Reply:
updated!